gigascience/gigadb-website
# New feature/1553 fix outstanding accessibility issues in admin pages

# Pull request for issue: #1553

This is a pull request for the following functionalities:

* Fix small scope outstanding accessibility and layout issues in admin pages (large scope issues would be fixed in separate tickets)
* Update playwright tests to more easily run subsets of tests, based on tags

## How to test?

* Run playwright tests for current tag (see playwright/tags.js), i.e.
```sh
cd playwright
npm i # if you didn't install deps
npm test
```
* Verify implementations described below

## How have functionalities been implemented?

* Added horizontal scroll to wide tables (http://gigadb.gigasciencejournal.com:9170/attribute/admin)

![image](https://github.com/gigascience/gigadb-website/assets/143437854/72359a4e-43d0-4e83-b98f-f83f8d331f09)

* Set action buttons in a row for wide tables (http://gigadb.gigasciencejournal.com:9170/attribute/admin)
![image](https://github.com/gigascience/gigadb-website/assets/143437854/4fc2882e-94b8-4e32-9b2c-edc110f183ef)

* Use html buttons for modal and add error status (in local dev, delete action fails)
![image](https://github.com/gigascience/gigadb-website/assets/143437854/de78e32d-4691-41cd-b9b5-984a003a557f)

* Update alert danger style to use brand color
![image](https://github.com/gigascience/gigadb-website/assets/143437854/9ca8a67a-e821-40da-8213-42aec62469c1)

* "merge..." button only displayed in update page as it requires an existing id
![image](https://github.com/gigascience/gigadb-website/assets/143437854/131b11bb-0530-4e22-af51-71ee908a29b3)

* Fix some of the issues in http://gigadb.gigasciencejournal.com:9170/adminDataset/update/id/5 / http://gigadb.gigasciencejournal.com:9170/adminDataset/create

  * Fix broken aria reference
  * Use CustomGridView component (i.e. styled pagination and action buttons)
![image](https://github.com/gigascience/gigadb-website/assets/143437854/48ca2c2d-7a44-4635-876b-7a8784ab9995)
  * Updated aria label from "delete file" to "delete image file"  to provide a bit more context
![image](https://github.com/gigascience/gigadb-website/assets/143437854/87160172-1477-43cb-bc79-ddab3ef16176)
  * "Mint DOI" is a button and message is an alert (so it gets announced by screen readers)  
![image](https://github.com/gigascience/gigadb-website/assets/143437854/968b8fc4-f722-4c65-81e9-8fba8380192b)

* Updated curator log pages to modern layout:
  * http://gigadb.gigasciencejournal.com:9170/curationLog/view/id/3
  * http://gigadb.gigasciencejournal.com:9170/curationLog/update/id/3

* http://gigadb.gigasciencejournal.com:9170/curationLog/create/id/22 Updated breadcrumbs so they link to dataset page, by grabbing id from pathname (which for the create page, is the dataset id, not the curator log id)

* Revamped playwright tests with a tag config to make it easier to run subsets of tests. The issue is that running the entire test suite locally takes too many resources and tends to freeze the (my) computer. With the tags, it is possible to easily run the tests associated to a specific ticket, by ticket number

## Any issues with implementation?

* http://gigadb.gigasciencejournal.com:9170/adminDataset/create

There is a duplicate element here:
![image](https://github.com/gigascience/gigadb-website/assets/143437854/9114fe6a-50b0-4fd1-898a-141edbdce400)
I suspect this is due to a duplicate in the mock DB used for local development. Something to keep an eye out to make sure it doesn't happen in production

## Any changes to automated tests?

None

## Any changes to documentation?

None

## Any technical debt repayment?

None

## Any improvements to CI/CD pipeline?

None
